Added summary on warnings and errors (#2581)

bladerunner2020 opened the pull request:
Added a summary message about warnings and errors if any (see #2581) in the following functions:
- generateDocs
- generateJson

Added tests.

Gerrit0 commented on src/lib/application.ts:
```diff
@@ -525,6 +525,14 @@ export class Application extends ChildableComponent<
         out = Path.resolve(out);
         await this.renderer.render(project, out);
+
+        // Generate a message summarizing the number of errors and warnings if there are any
```
I think this is the wrong place to put this. TypeDoc has a `--watch` option, which will result in `generateDocs` and `generateJson` being called many times. If a user has both options turned on, then with this implementation they may get two warnings/errors for each compilation.

I think the way I'd prefer to handle this is to add a public `logRunSummary` (better name?) method on `Application` which is called in `cli.ts` in the appropriate places

Gerrit0 replied:
This also has the benefit of making the tests for this feature much less hacky - I've very intentionally avoided mocking pieces of TypeDoc in tests (almost) everywhere so far, they tend to lead to testing implementation details+easily broken tests

bladerunner2020 replied:
> I think this is the wrong place to put this. TypeDoc has a `--watch` option, which will result in `generateDocs` and `generateJson` being called many times. If a user has both options turned on, then with this implementation they may get two warnings/errors for each compilation.
> 
> I think the way I'd prefer to handle this is to add a public `logRunSummary` (better name?) method on `Application` which is called in `cli.ts` in the appropriate places

That seems reasonable. I'll take a look at it.

Gerrit0 reviewed (changes requested):
Thanks for the PR! I have a couple concerns

bladerunner2020 commented:
I moved `logRunSummary` to `cli.ts` and agree that it's a better place for it. I also removed the test file as the tests are no longer necessary due to the simplicity of this solution. However, I did not add `logRunSummary` to the Application; I left it as a function in `cli.ts`.

Gerrit0 commented:
Thanks!
